test/utils/k8s/job.go

```diff
@@ -175,22 +176,50 @@ func AssertJob(t *testing.T, job *batchv1.Job) {
 }
 
 func GetJobLogs(ns string, kubeClient kubernetes.Interface, name string) (string, error) {
+	return GetJobsLogs(ns, kubeClient, name, false)
+}
+
+// Returns the logs of the pods related to a given job.  If 'all' is true, all runs will be
+// returned; failures to get individual logs will be reported on the output, but not cause
+// the function to fail.
+//
+// If 'all' is false, only the last run's logs will be returned.
+func GetJobsLogs(ns string, kubeClient kubernetes.Interface, name string, all bool) (string, error) {
```

Comment:
It might be better to return a map indexed by pod name with value being the logs,
in the former `GetJobLogs` you keep it as is and just consume the first element of
the returned map.

I believe it might work for the long term as future jobs can, eventually, save each log
to an individual file.

Reply:
If map indexed by podname, we lose the timing information...  It could be a slice of structs that have the logs + meta info (timing, podname, etc), and already in the proper order, instead.

On your suggestion, I'm leaning toward creating a third function, then, that returns the map or slice.  GetJobLogs would do as  you say, and GetJobsLogs would call this new function and return the logs as a string, just like GetJobsLogs does today.  That would make that preparation, but still require little changes on the job itself.

On this PR, I just want to get proper information for my investigation; I wanted to make as little changes to the test as possible, before I have root cause for the issue I'm investigating.

Reply:
Sounds fair.